### PR TITLE
[workflow]: fix the source-url

### PR DESCRIPTION
**Pull request by soumyaDghosh**
This should fix the issue, tested locally

image

### CODE REVIEW

File: .github/workflows/sync-upstream.yml [hunk 1]
```diff
@@ -24,6 +24,6 @@ jobs:
           update-script: |
             source_url="$(curl -sL https://developer.android.com/studio/index.html | grep -Eo '"((https)?://.*linux.tar.gz)"' | tr -d '"')"
             codename="$(curl -sL https://developer.android.com/studio/index.html | grep -Po 'Download Android Studio (?!today)[A-Za-z0-9]+' | head -n1 | cut -d ' ' -f4)"
-            version="$(echo "$source_url" | rev | cut -d '/' -f2 | rev && echo "$codename")"
-            sed -i 's/^\(source: \).*$/\1'"\"$source_url\""'/' snap/snapcraft.yaml
+            version="$(basename "$source_url" .tar.gz | grep -oP '\d+\.\d+\.\d+\.\d+') $codename"
+            yq -i ".parts.android-studio.source = \"$source_url\"" snap/snapcraft.yaml
```
Comment (jnsgruk) [inline .github/workflows/sync-upstream.yml]: Please revert this change to `yq` - this will mutate and reformat the whole yaml file each time it's run, and not necessarily for the better. `sed` was likely a good choice here.

**Review by jnsgruk — CHANGES_REQUESTED** [review]
Version wrangling looks good, just one comment :+1:

File: .github/workflows/sync-upstream.yml (hunk 1 above)
Comment (soumyaDghosh) [reply .github/workflows/sync-upstream.yml]: `sed` is not working with `url`s as you can see in this [discussion](https://github.com/snapcrafters/android-studio/pull/118). That's why this PR is here to change it. It's getting confused with the `/` in the `url`s. I tried the same command, and it works if it doesn't contain `/`.

Comment (jnsgruk) [reply .github/workflows/sync-upstream.yml]: You can use a different separator with `sed` which I think will solve it. Something like `s|foo|bar`. I've not tested, but the original would then look something like this:

```suggestion
            sed -i 's|^\(source: \).*$|\1'"\"$source_url\""'|' snap/snapcraft.yaml
```

Comment (soumyaDghosh) [reply .github/workflows/sync-upstream.yml]: Let me test it then

Comment (soumyaDghosh) [reply .github/workflows/sync-upstream.yml:28]: `sed -i 's|^\(source: #).*$|"\"$source_url\"|"' snap/snapcraft.yaml`
`sed -i 's#^\(source: #).*#"\"$source_url\"#"' snap/snapcraft.yaml`
`sed -i 's#^\(source: \).*$#\1'"\"$source_url\""'#' snap/snapcraft.yaml`

tried all three above, didn't work

Comment (soumyaDghosh) [reply .github/workflows/sync-upstream.yml]:
```
sed -i 's|^\(source: \).*$|\1'"\"$source_url\""'|' snap/snapcraft.yaml
sed -i 's#^\(source: \).*$#\1'"\"$source_url\""'#' snap/snapcraft.yaml
sed -i 's#^\(source: #).*#"\"$source_url\"#"' snap/snapcraft.yaml
sed -i 's|^\(source: #).*$|"\"$source_url\"|"' snap/snapcraft.yaml
```
tried all of these, but none worked. :-(

image

Comment (jnsgruk) [reply .github/workflows/sync-upstream.yml]: The following works:

```
sed -i "s|source: .*$|source: $source_url|g" snap/snapcraft.yaml
```

Comment (soumyaDghosh) [reply .github/workflows/sync-upstream.yml]: Thanks! Done!

**Review by jnsgruk — APPROVED** [review]

**Comment by jnsgruk** [comment]
Nice, looking sharp! Thanks!
